components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/firewallchecktrigger/firewallchecktrigger.go

```diff
@@ -0,0 +1,266 @@
+package firewallchecktrigger
+
+import (
+	"net/http"
+
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/constants"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/models"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger"
+	"github.com/chef/automate/lib/logger"
+)
+
+var (
+	BastionNodeIP = "127.0.0.1"
+)
+
+type FirewallCheck struct {
+	host string
+	port string
+	log  logger.Logger
+}
+
+func NewFirewallCheck(log logger.Logger, port string) *FirewallCheck {
+	return &FirewallCheck{
+		log:  log,
+		host: constants.LOCAL_HOST_URL,
+		port: port,
+	}
+}
+
+func (fc *FirewallCheck) Run(config models.Config) []models.CheckTriggerResponse {
+	fc.log.Info("Performing Firewall check from batch check ")
+	requests := makeRequests(config)
+	return trigger.RunCheckMultiRequests(config, fc.log, fc.port, constants.FIREWALL_API_PATH, "", http.MethodPost, requests)
+}
+
+// The request response is being constructed based on the https://docs.chef.io/automate/ha_on_premises_deployment_prerequisites/#firewall-checks (Firewall Checks)
+func makeRequests(config models.Config) []trigger.ReqBody {
+	var reqBodies []trigger.ReqBody
+
+	// _____________________________________________ Row 1: Chef Automate to all the OS and PG nodes _____________________________________________
+	// Dest postgres
+	for _, sourceNodeIP := range config.Hardware.AutomateNodeIps {
+		for _, destNodeIP := range config.Hardware.PostgresqlNodeIps {
+			reqBody := trigger.ReqBody{
+				NodeType:                   constants.AUTOMATE,
+				SourceNodeIP:               sourceNodeIP,
+				DestinationNodeIP:          destNodeIP,
+				DestinationServicePort:     "7432",
+				DestinationServiceProtocol: "tcp",
+				Cert:                       config.Certificate.Nodes[0].Cert,
+				Key:                        config.Certificate.Nodes[0].Key,
```

Comment:
How can we assign Key for now I'm taking the 0th index but we can modify it?

cc: @Dmaddu